GoogleForCreators/web-stories-wp
# Use Moveable in Library for Media

## Summary
Uses Moveable in Library instead of default dragging.

EDIT: Snapping will be addressed separately!
This is currently used for images and videos only -- shapes and text sets will be handled separately.


## Relevant Technical Choices
- Adds an overlay to `Editor` which will contain the Moveable that's being dragged.
- Adds a clone image that will be added to the DOM if the user hovers on media -- this clone will then be dragged. We can't move the original element itself since it wouldn't be visible -- it would be under the Workspace layer.
- Adds a frame with the size of the element in the library that will be the target for the Moveable and dragged together with the clone. We need this for the snapping to work.


## To-do

- [x] Fix base-color calculation for new dragging.
- [x] Update tests where necessary



## User-facing changes



## Testing Instructions
I
1. Add an image or a video.
2. Now drag an image or video to the canvas
3. While dragging verify that it displays information about drop targets
4. Verify that dropping into the existing image/video is possible

II
1. Drag a GIF to the canvas, verify the gif is animating while dragging.

III
1. Verify that dragging image/video from both local/3rd party media work.




---



Fixes #5335 
Fixes #5057 
Fixes #1930 
~Fix for #1979~  will be implemented separately!
